api.planx.uk/modules/auth/middleware.ts

```diff
@@ -16,7 +16,9 @@ export const userContext = new AsyncLocalStorage<{ user: Express.User }>();
 /**
  * Validate that a provided string (e.g. API key) matches the expected value
  */
-const isEqual = (provided = "", expected: string): boolean => {
+export const isEqual = (provided?: string, expected?: string): boolean => {
+  if (!provided || !expected) throw Error("Values must be set for isEqual()");
```

Comment:
An error is thrown without this addition by `hash.copy().update()` which requires a string.

Setting this here just allows us a slightly nicer error message and we can pass through `string | undefined` as an argument (e.g. `req.headers.authorization` or `process.env.SOMETHING`).